### ISSUE TITLE
Usability of menus on small screens

### ISSUE THREAD

**Issue by skaldarnar** [MovingBlocks/Terasology#2585]
### Description

Navigating the menus can become tricky on devices with smaller screens/resolutions. The snippet below shows the create game screen on my laptop. As can be seen, I cannot select the world generator type properly (the game windows hits the bottom end of the screen).

![out](https://cloud.githubusercontent.com/assets/1448874/19890581/f998efe6-a03b-11e6-8d25-c3f3a908639d.gif)

### Definition of 'Done'
- [ ] all menus are properly usable on both big and small screens/resolutions

### Where to start?
- Check out the [NUI Quick Start](https://github.com/Terasology/TutorialNui/wiki/Quick-Start) tutorial
- Familiarize yourself with the game menu screens, especially [create game screen](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/resources/assets/ui/menu/createGameScreen.ui)
- Look at [videoMenuScreen.ui](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/resources/assets/ui/menu/videoMenuScreen.ui) to understand the use of `ScrollableArea`

**Comment by Cervator** [comment]
Old similar issue over at #1627 with some more comments.

We may be looking at two different things here:

* The Create Game screen is getting cluttered. Not *everything* necessarily needs to go inside the framed box and/or we could make it wider (then labels could go on the same line as their interactive widget in at least some cases). Like the sneaky developer tools button hiding in the bottom right corner on the in-game menu maybe we should have secondary information in the main menu like save path in a corner rather than in the main frame
* The frame box itself should become scrollable if needed. However that may not fix drop-downs appearing below the game window. Old issue #844 touched on that briefly and suggested a drop-up (naming?) could be used instead if the widget is too close to the bottom of the window.

**Comment by kartikey0303** [comment]
![screenshot 149](https://cloud.githubusercontent.com/assets/14962635/20134081/1027e0dc-a690-11e6-82bb-2053aed5f512.png)
![screenshot 150](https://cloud.githubusercontent.com/assets/14962635/20134082/102b2af8-a690-11e6-9748-0c622799019a.png)

**Comment by skaldarnar** [comment]
@kartikey0303 The scrollable area adds a lot of value to it. If the "dropup" thing is a bit tricky, I'd be happy to merge it with the applied changes right now and leave a note for later.

**Comment by kartikey0303** [comment]
@skaldarnar i have made an extra class for UIDropupScrollable and added it to the last drop down of world generator. Currently it is not smart to detect if it should open up or down. How do we calculate the distance from the edge? If i can figure that out maybe we can add the same functionality without a new class. Currently the dropdown in case of World generator always open up because if it opens down then it goes beyond the scrollable area.

**Comment by Cervator** [comment]
Bump - hey @kartikey0303 thanks for the work on this, got it online somewhere yet? I would love to get this merged in soon since it would look pretty bad to do a stable release with UI elements jumbled up right on the Create Game screen. Way better if we can fix this first :-)

@skaldarnar Any other thoughts on this one? And dragging in @rzats as well because UI :-)

**Comment by rzats** [comment]
@Cervator: I left some feedback on this in Slack earlier.

tl;dr: needs some margins (especially on the left), dropup/dropdown switching should be implemented for `UIDropdownScrollable` and, if it doesn't take too much effort, `UIDropdown` as well.

**Comment by kartikey0303** [comment]
Yeah I did the left margin thing. There was less margin because an extra scroll bar was added. Just needed to increase the width of the box. I will also make a semi-transparent like it was earlier. I think the opaque property is because of the scrollable area. Considering the smart dropup and down, it may take some time as I have my exams going on and I am not getting time to fix anything. I will probably get some time day after tomorrow, then  will do the smart dropdown.

**Comment by starlord1311** [comment]
i would like to take up this issue

**Comment by starlord1311** [comment]
i would like to take up this issue

**Comment by starlord1311** [comment]
can someone help me with it?

**Comment by jellysnake** [comment]
Hey @starlord1311 so from reading the issue discussion it seems that @kartikey0303 is currently working on this issue.

**Comment by kartikey0303** [comment]
Hello @starlord1311. As you can see in my [comment](https://github.com/MovingBlocks/Terasology/issues/2585#issuecomment-259374087) above. I added the scrollable UI element to the earlier to the code. However there was a problem with one of the dropdowns. As the UI won't render anything outside the box, when the dropdown was opened there was no way to see the contents of the dropdown on a smaller screen. You can try hard coding the dropdown to open as I did and merge it. I achieved it by creating a new UI element "dropup" which opens up instead of down. For more reference you can consult @skaldarnar's [comment above](https://github.com/MovingBlocks/Terasology/issues/2585#issue-186530670).

Ideally the UI element dropdown should render outside of the scrollable area and it. It should be smart enough to drop down or up according to the screen size.

But for starting you can go through the references mentioned above by @skaldarnar.
